app/javascript/lib/web-worker.js

```diff
@@ -0,0 +1,78 @@
+/** Web worker that wraps a CPU-intensive function */
+function setViolinCellIndexesWorker() {
```

Comment:
Very cool to see this setup - took me a minute to grok but I _think_ I understand how this works.  Question: are there other things in the UI where this paradigm could be applied to improve UI performance?

Reply:
Potentially!  Web workers enable client-side multithreading.  Generally, I think, that's is most valuable when A) there's a compute-intensive task, B) it takes a while to complete, and C) completing it as fast as possible isn't a high priority for the UX.

* **Client-side file validation for big local files**.  Right now the UI doesn't freeze, but it does sputter; web workers would make it smooth.  In this case, the cost of the slow-down for web workers is probably worth the more complete validation we could do for CSFV.

* **Cell filtering**.  Applying filters to large numbers of cells can sometimes take a while, and that can make the UI less responsive than usual.  Computing filter updates in a non-UI thread would help that.  However, while it would keep the UI responsive, it might actually make filtering itself a bit slower than doing it on the main UI thread.  That's at least what I observed when experimenting with web workers and CPU-intensive work in other context a year or two ago.